### ISSUE TITLE
Create_scoped_session fails on latest flask/werkzeug 2.1.x version.

### ISSUE THREAD

**Issue by CaBazaga** [osohq/oso#1548]
New recently released Werkzeug version 2.1.x, internally used by Flask, finally removed the deprecated `__ident_func__ ` property

Werkzeug 2.0.x old code with deprecation warning:
<https://github.com/pallets/werkzeug/blob/aeb3935fbf8a11bcb120e188e443e3432e1879ff/src/werkzeug/local.py#L133>

This lead to the following OSO line:
<https://github.com/osohq/oso/blob/026902c1b8b4b087bf6c1154b9e2a8548dc8bd53/languages/python/sqlalchemy-oso/sqlalchemy_oso/flask.py#L50>

Fails as follows:

```  
File ".../app/database/base.py", line 12, in <module>
    OsoDB = AuthorizedSQLAlchemy(
File ".../python3.10/site-packages/sqlalchemy_oso/flask.py", line 44, in __init__
    super().__init__(**kwargs)
File ".../python3.10/site-packages/flask_sqlalchemy/__init__.py", line 753, in __init__
    self.session = self.create_scoped_session(session_options)
File ".../python3.10/site-packages/sqlalchemy_oso/flask.py", line 50, in create_scoped_session
    scopefunc = options.pop("scopefunc", _app_ctx_stack.__ident_func__)
AttributeError: 'LocalStack' object has no attribute '__ident_func__'
```

After a small research and test it seems removing the offending lines:

<https://github.com/osohq/oso/blob/026902c1b8b4b087bf6c1154b9e2a8548dc8bd53/languages/python/sqlalchemy-oso/sqlalchemy_oso/flask.py#L50>

<https://github.com/osohq/oso/blob/026902c1b8b4b087bf6c1154b9e2a8548dc8bd53/languages/python/sqlalchemy-oso/sqlalchemy_oso/flask.py#L55>

Solved the problem, but I'm not sure if it may have some side effects.

In any case its returned value (the get ident function) was as defined on:
<https://github.com/pallets/werkzeug/blob/aeb3935fbf8a11bcb120e188e443e3432e1879ff/src/werkzeug/local.py#L19>
as:
```python
try:
    from greenlet import getcurrent as _get_ident
except ImportError:
    from threading import get_ident as _get_ident
```

**Comment by balaviku** [comment]
I also faced same issue.. Instead of removing the offending lines I added the werkzeug== 2.0,3 before installing flask in my requirement.txt and it worked!!!

**Comment by gkaemmer** [comment]
Thanks for the detailed report @CaBazaga -- I opened up a PR to fix here: #1559

**Comment by MT-Cash** [comment]
The fix was werged on 29th of April. I guess it was part of the Werkzeug 2.2.0 release.
